python/ray/util/sgd/torch/torch_trainer.py

```diff
@@ -150,7 +146,6 @@ def __init__(
             timeout_s=NCCL_TIMEOUT_S,
             use_fp16=False,
```

Comment:
Should we move `use_fp16` also into `register`?

Reply:
I think it's fine to leave it as is. Currently the TorchRunner needs to be aware of this to know whether to save/load state from amp, so it needs to be passed in via the TorchTrainer.